### ISSUE TITLE
support custom CAs and ca-bundle from openshift Cluster Proxy Config in AWS S3 connectivity 

### ISSUE THREAD

**Issue by shalberd** [opendatahub-io/modelmesh-serving#113]
In ODH Dashboard Model Serving, there is the possibility to define access (url, credentials and so on) to S3-compatible storage buckets for model files (described in ODH Dashboard as Data Connection).

If those files were located an a server that has certificates based on custom / private PKI CAs, there can be SSL trust validation issues when connecting to e.g. ceph or IBM object storage via https.

Allow mounting-in of trusted-ca-bundle cert trust into modelmesh serving via AWS_CA_BUNDLE env var, in addition to HTTP_PROXY and HTTPS_PROXY and NO_PROXY info from OCP cluster proxy resource.

https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#environment-variable-configuration

**Describe alternatives you have considered**

**Additional context**

https://github.com/opendatahub-io/odh-dashboard/issues/1381#issuecomment-1594587169

Similar to, but other place to inject most likely, this effort in notebooks:

https://github.com/opendatahub-io/kubeflow/pull/43

**Comment by shalberd** [comment]
conceptually related to opendatahub-io/kubeflow/issues#105

**Comment by Jooho** [comment]
@shalberd I think this issue would be a workaround to solve this AWS_CA_BUNDLE issue.
https://github.com/opendatahub-io/odh-model-controller/pull/62

In addition, we are discussing this cert issue with [this ticket](https://github.com/opendatahub-io/odh-model-controller/issues/61) so please check it out.

**Comment by bdattoma** [comment]
Hello, there are 3 issues at least which sound aiming for the same result:
https://github.com/opendatahub-io/modelmesh-serving/issues/113 
https://github.com/opendatahub-io/modelmesh-serving/issues/148
https://github.com/opendatahub-io/odh-model-controller/issues/61

Could you please review them and close duplicates? @Jooho @shalberd

**Comment by shalberd** [comment]
would this be a replacement for or an extension to the work done in #362?

**Comment by gregsheremeta** [comment]
> The advantage would be that I only have to add custom trusted CAs and self-signed certs in PEM format once at central cluster Proxy config in a secret in namespace openshift-config, which is much more streamlined and less distributed.

Long term, we are pursuing a global cluster-level approach. Whether that uses the Proxy or not is TBD.

A future global cluster-level approach does not obviate the need for component-level control as well.

**Comment by shalberd** [comment]
agreed. see my thoughts [elsewhere](https://github.com/opendatahub-io/data-science-pipelines-operator/pull/440#issuecomment-1799074925)

**Comment by heyselbi** [comment]
@shalberd is there outstanding issue specifically for us since this is more a platform-level issue?

**Comment by spolti** [comment]
Hi, is it still an issue?
